app/models/concerns/sufia/generic_file/characterization.rb

```diff
@@ -0,0 +1,120 @@
+module Sufia
+  module GenericFile
+    module Characterization
+      extend ActiveSupport::Concern
+      included do
+        contains "characterization", class_name: 'FitsDatastream'
+        property :mime_type,         delegate_to: 'characterization', multiple: false do |index|
+          index.as :stored_searchable
+        end
+        property :format_label,      delegate_to: 'characterization'
+        property :file_size,         delegate_to: 'characterization'
+        property :last_modified,     delegate_to: 'characterization'
+        property :filename,          delegate_to: 'characterization' do |index|
+          index.as :stored_searchable
```

Comment:
N.B. we're overriding this file from Sufia so that we can index the filename to keep download link rendering fast with Peter's changes